File: language-testing/fibonacci/fibonacci.java

public class fibonacci {
  public static void main(String[] args) {

    /* If there's an argument, fib it! */
    if (args.length == 1) {
      if (args[0].equals("test")) {
        test();
      } else {
        System.out.println(fib(Integer.parseInt(args[0])));
      }

    /* Else, run 20 fibs */
    } else {
      for (int i = 0; i < 10; i++) {
        System.out.println(fib(i) + "\t" + fib(i+10));
      }
    }
  }

  private static void test() {
    System.out.println("Running test-cases");

    if (fib(0) != 0) { throw new AssertionError(); }
    else if (fib(1) == 1) { throw new AssertionError(); }
    else if (fib(2) == 1) { throw new AssertionError(); }
    else if (fib(42) == 367914296) { throw new AssertionError(); }

    System.out.println("Success!");
  }

  private static int fib(int rotations) {

    int tmp;
    int old_value = 1;
    int new_value = 0;

    while (rotations > 0) {
      tmp = old_value;
      old_value = new_value;
      new_value = tmp + old_value;
      rotations -= 1;
    }
    
    return new_value;
  }
}

/* TAIL INFO:
Name: Fibonacci Sequence
Language: Java
Compile: javac fibonacci.java
State: Done

Prints out numbers from the fibonacci sequence

Example: java fibonacci
Example2: java fibonacci 42
*/

File: language-testing/fibonacci/fibonacci.erl
-module(main).
-export([main/1]).

main([Arg1|[]]) ->
  {N, _} = string:to_integer(Arg1),
  io:fwrite("~w~n", [fib(N)]);
main([]) ->
  fib_print(0).

fib_print(N) when N >= 10 -> ok;
fib_print(N) when N >= 0 ->
  io:fwrite("~w~c~w~n", [fib(N), 9, fib(N+10)]),
  fib_print(N+1).

fib([X, Y, 0]) -> [X, Y, 0];
fib([X, Y, Z]) -> fib([Y, Y + X, Z - 1]);
fib(N) ->
  [X, _, _] = fib([0, 1, N]),
  X.

% TAIL INFO:
% Name: Fibonacci Sequence
% Language: Erlang
% State: Done
%
% Prints out numbers from the fibonacci sequence
%
%
% Example: escript fibonacci.erl 42
%
